core/src/main/java/jenkins/security/ResourceDomainRootAction.java

```diff
@@ -165,7 +167,7 @@ private static String getResourceRootUrl() {
     @CheckForNull
     public Token getToken(@Nonnull DirectoryBrowserSupport dbs, @Nonnull StaplerRequest req) {
         // This is the "restOfPath" of the DirectoryBrowserSupport, i.e. the directory/file/pattern "inside" the DBS.
-        final String dbsFile = req.getRestOfPath();
+        final String dbsFile = req.getOriginalRestOfPath();
```

Comment:
This is the most important part of the fix, otherwise the math below will be off: We subtracted the length of `foo bar` from a string ending in `whatever/foo%20bar`…